presto-main/src/main/java/com/facebook/presto/operator/repartition/DecodedBlockNode.java

```diff
@@ -42,10 +43,16 @@
     private final Object decodedBlock;
     private final List<DecodedBlockNode> children;
 
-    public DecodedBlockNode(Object decodedBlock, List<DecodedBlockNode> children)
+    // The portion of the size of the exclusive content of this block to the total size. For example, the exclusive content
+    // of an ArrayBlock is its offsets and nulls array while not its elementBlock. It is used to estimate the limit of the
+    // buffers' sizes.
+    private double sizePortionInPage;
```

Comment:
Can this be calculated within this class as getRetainedSizeInBytes - sum(child.getRetainedSizeInBytes())? If not, perhaps, move this change into a separate commit.

Reply:
@mbasmanova Yes you're actually right. I moved the logic to DecodedBlockNode and made a new commit `52456a0fec Introduce sizePortionInPage in DecodedBlockNode`